src/adagrams.js

```diff
@@ -1,15 +1,144 @@
+const letterPool = {
+  A: 9,
+  B: 2,
+  C: 2,
+  D: 4,
+  E: 12,
+  F: 2,
+  G: 3,
+  H: 2,
+  I: 9,
+  J: 1,
+  K: 1,
+  L: 4,
+  M: 2,
+  N: 6,
+  O: 8,
+  P: 2,
+  Q: 1,
+  R: 6,
+  S: 4,
+  T: 6,
+  U: 4,
+  V: 2,
+  W: 2,
+  X: 1,
+  Y: 2,
+  Z: 1,
+};
+
 export const drawLetters = () => {
   // Implement this method for wave 1
+  const playersHand = [];
+  const otherLetters = [];
+  const copyOfLetterPool = {...letterPool};
+  for (let letter in copyOfLetterPool) {
+    for (let i = 0; i < copyOfLetterPool[letter]; i++) {
+      otherLetters.push(letter);
+    }
+  }
+  for (let i = 0; i < 10; i++) {
+    let index = Math.floor(Math.random() * otherLetters.length);
+    // add to playersHand
+    playersHand.push(otherLetters[index]);
+    // remove from other letters
+    otherLetters.splice(index, 1);
+  }
+  return playersHand;
 };
 
 export const usesAvailableLetters = (input, lettersInHand) => {
   // Implement this method for wave 2
+  const lengthOfHand = input.length;
+  let lettersInHandCopy = [...lettersInHand];
+  for (let i = 0; i < lengthOfHand; i++) {
+    let letter = input[i];
+    if (lettersInHandCopy.includes(letter)) {
+      let j = lettersInHandCopy.indexOf(letter);
+      lettersInHandCopy.splice(j, 1);
```

Comment:
The time complexity of splice in JS is the same as it is in Python, linear. Instead of splicing the letter in the loop, you could change the value of letter to 0 or some value that isn't in the alphabet.

Then when you check if lettersInHandCopy.includes(letter), you won't double count anything because "A", for example, would be changed to 0 (instead of removing it). That would get around the linear time complexity of splice